uniffi_bindgen/src/bindings/python/templates/FfiConverters.py

```diff
@@ -0,0 +1,607 @@
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(1, ">b")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(1, ">b", v)
+
+{% when Type::UInt8 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(1, ">B")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(1, ">B", v)
+
+{% when Type::Int16 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(2, ">h")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(2, ">h", v)
+
+{% when Type::UInt16 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(2, ">H")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(2, ">H", v)
+
+{% when Type::Int32 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(4, ">i")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(4, ">i", v)
+
+{% when Type::UInt32 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(4, ">I")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(4, ">I", v)
+
+{% when Type::Int64 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(8, ">q")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(8, ">q", v)
+
+{% when Type::UInt64 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(int)
+    lower = staticmethod(int)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(8, ">Q")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(8, ">Q", v)
+
+{% when Type::Float32 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(float)
+    lower = staticmethod(float)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(4, ">f")
+
+    @staticmethod
+    def write(builder, v):
+        builder = builder._pack_into(4, ">f", v)
+
+{% when Type::Float64 -%}
+class {{ typ|ffi_converter_name }}:
+    lift = staticmethod(float)
+    lower = staticmethod(float)
+
+    @staticmethod
+    def read(stream):
+        return stream._unpack_from(8, ">d")
+
+    @staticmethod
+    def write(builder, v):
+        builder._pack_into(8, ">d", v)
```

Comment:
I considered creating a base class that handled the common parts of these, but I went with a flat hierarchy because I was a paranoid about performance.  We're going to be using these everytime we want to do anything with an argument/return value, so I figured we should keep the number of lookups to a minimum.

On the topic of performance.  One negative about this system is that we always have a `FfiConverter` call.  Before, for simple types like this we would just call `int(x)`.  We could consider adding back the lift_py filter function, but have it only handle optimizations like that.

Reply:
I don't really have a strong opinion here, and given we don't really rely on Python bindings, I'd be fine with a comment about the optimization. What I still can't quite picture here is what this means for the strongly-typed bindings I we should probably start focusing on (say) kotlin to make sure we aren't painting ourselves into a dynamically-typed corner.